### PR TITLE
A few small improvements

### PR DESCRIPTION
**Pull request by KKV9**
Feat: --block and --confirm flags to better emulate the default yazi shell.
Feat: Capitalise shell prompt.
Feat: Support for nushell.
Feat: Accept "Kornshell" or "ksh" as arguments.
Feat: Use default shell and not blocking by default.
Feat: Update LICENSE and README.
Refactor: General formatting.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,46 +37,72 @@ There are 2 ways you can set your custom-shell.
...
 
 ```toml
 [[manager.prepend_keymap]]
-on = [ ":" ]
-run = "plugin custom-shell --args=zsh" # Example
-desc = "custom-shell as default"
+on = [ ";" ]
+run = "plugin custom-shell --args=zsh"
+desc = "custom-shell with zsh"
+```
```
Comment (AnirudhG07) [inline README.md:80]: Dont change the example

File: README.md [hunk 2]
```diff
@@ -37,46 +37,72 @@ There are 2 ways you can set your custom-shell.
...
+desc = "Run ls table in nu sort by ...?"
+```
+
+To run a specific command in a specific shell, you can set the keymappings as-
+
+```toml
+[[manager.prepend_keymap]]
+on = [ ";" ]
+run = "plugin custom-shell --args='fish \"echo example command with --block and --confirm flags ; read c\" --block --confirm'"
+desc = "Run a blocking echo command with fish"
```
Comment (AnirudhG07) [inline README.md:97]: I dont like the idea to assign `;` with fixed commands as keybindings. They should input the command. But I have alternative for this. Ill tell

**Comment by AnirudhG07** [comment]
I have an idea if you are thinking to implement special commands with keybindings. For example, let's keep `:` and `;` as it is with `plugin custom-shell` and with `--block` as done. I use `';` for custom-shell, so like this, we can let the users configure special commands with ["'", "char"]. Like `'l` will run say `ls -la`, etc.
Also shorten --block and --confirm, if we are checking for it anyway, why not make it more convenient to add.

**Comment by AnirudhG07** [comment]
Configure chars from 0-9,a-z seems enough. Keep the custom-shell current as it is with `';`, with special commands can be configured for other keys.

**Comment by AnirudhG07** [comment]
The codes looks fine to me

File: README.md (hunk 1 above)
Comment (KKV9) [reply README.md:80]: The example was changed because the behaviour was changed. The shell no longer defaults to blocking.

File: README.md (hunk 2 above)
Comment (KKV9) [reply README.md:97]: These are examples. I doubt anyone will use this keybind.

**Comment by KKV9** [comment]
> The codes looks fine to me

 I tested it locally and it works fine.

**Comment by AnirudhG07** [comment]
ill suggest some fixed changes to do in the readme before merging the changes, within a day.

**Comment by AnirudhG07** [comment]
Please make the below changes
1) Let --confirm be true by default. Keep block available for true/false. 
2) In the code. let block be -b and confirm be -c. ~Also add --interactive after `shell` compulsorily since v0.3 has all its keymap.toml shell codes with --interactive.~ Apparently you cant have interactive and confirm at the same time... my bad
2) I'll change the readme myself according to the new codes. So no need to push the readme.

**Comment by KKV9** [comment]
Done

**Comment by AnirudhG07** [comment]
Ok now it doesnt work for me at all. any of the commands in your fork's readme where the prompt is supposed to come, is not coming.

**Comment by KKV9** [comment]
Correct. When --confirm is enabled, there is no prompt.

**Comment by AnirudhG07** [comment]
I am trying to fix it. Maybe have to change many things. If one thing, works other breaks. I'll respond in a day. If you solve it. Share the code, instead of a commit. The new version of yazi does not support confirm and interactive together. By this I want to refer, keeping value = value_string is showing that error (after some tweaks I made).

**Comment by KKV9** [comment]
This behaviour is expected. set --confirm to false to see a prompt

**Comment by AnirudhG07** [comment]
Hey, I have been looking into various options myself, and it seems to make it generalised and stuff. Your original codes seemed great(sorry for the troubles). Also I have also looked into your fork, some more additions have been added. If you feel like creating a new PR, you are welcome to. Thanks

**Pull request closed without merging**